[BUGs]自定义称号

复现步骤：
1.使用称号商店查看可购买的称号（1-小富翁,10000金币 2-测试用的称号2,2金币 3-测试用的称号1,1金币）
2.购买称号 测试用的称号1
3.切换称号 1
4.使用info查看签到信息报错 [bug1]
5.切换称号 0
6.使用info查看签到信息 金币-10000 [bug2]


以下是title.json中的内容：
[
    {
        "templateCode": "测试1",
        "validityPeriod": 365,
        "titleName": "测试用的称号1",
        "price": 1,
        "gradient": false,
        "impactName": false,
        "title": "[测试1]",
        "sColor": "#000000",
        "eColor": "#ffffff"
    },
    {
        "templateCode": "测试2",
        "validityPeriod": 365,
        "titleName": "测试用的称号2",
        "price": 2,
        "gradient": true,
        "impactName": false,
        "title": "[测试2]",
        "sColor": "#CD2990",
        "eColor": "#7FFFD4"
    }
]



以下是报错内容
2024-08-23 06:28:40 I/HuYanEconomy: 壶言经济--个人信息指令
2024-08-23 06:28:41 E/HuYanEconomy: 壶言经济--null
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at net.mamoe.mirai.internal.event.JvmMethodListenersInternal.registerEventHandler$callMethod$invokeWithErrorReport(JvmMethodListenersInternal.kt:148)
        at net.mamoe.mirai.internal.event.JvmMethodListenersInternal.access$registerEventHandler$callMethod$invokeWithErrorReport(JvmMethodListenersInternal.kt:26)
        at net.mamoe.mirai.internal.event.JvmMethodListenersInternal$registerEventHandler$callMethod$2.invokeSuspend(JvmMethodListenersInternal.kt:162)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
        at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
Caused by: java.lang.NumberFormatException: For input string: "#0" under radix 16
        at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
        at java.base/java.lang.Integer.parseInt(Integer.java:648)
        at java.base/java.lang.Integer.valueOf(Integer.java:963)
        at HuYanEconomy-1.4.0.mirai2.jar//cn.chahuyun.economy.utils.ImageUtil.hexColor(ImageUtil.java:165)
        at HuYanEconomy-1.4.0.mirai2.jar//cn.chahuyun.economy.entity.TitleInfo.getStartColor(TitleInfo.java:73)
        at HuYanEconomy-1.4.0.mirai2.jar//cn.chahuyun.economy.manager.UserManager.customBottom(UserManager.java:250)
        at HuYanEconomy-1.4.0.mirai2.jar//cn.chahuyun.economy.manager.UserManager.getUserInfoImageBase(UserManager.java:206)
        at HuYanEconomy-1.4.0.mirai2.jar//cn.chahuyun.economy.manager.UserManager.getUserInfoImage(UserManager.java:128)
        at HuYanEconomy-1.4.0.mirai2.jar//cn.chahuyun.economy.event.MessageEventListener.onMessage(MessageEventListener.java:87)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        ... 12 more

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -8,7 +8,7 @@ plugins {
 }
 
 group = "cn.chahuyun"
-version = "1.4.0"
+version = "1.4.1"
 
 repositories {
     mavenCentral()

--- a/src/main/java/cn/chahuyun/economy/HuYanEconomy.java
+++ b/src/main/java/cn/chahuyun/economy/HuYanEconomy.java
@@ -35,7 +35,7 @@ public final class HuYanEconomy extends JavaPlugin {
     /**
      * 全局版本
      */
-    public static final String VERSION = "1.4.0";
+    public static final String VERSION = "1.4.1";
     /**
      * 配置
      */

--- a/src/main/java/cn/chahuyun/economy/entity/title/CustomTitle.java
+++ b/src/main/java/cn/chahuyun/economy/entity/title/CustomTitle.java
@@ -31,4 +31,15 @@ public class CustomTitle {
         return new TitleTemplateSimpleImpl(templateCode, validityPeriod, titleName, price, gradient, impactName, title, sColor, eColor);
     }
 
+    /**
+     * 判断是否有空对象
+     *
+     * @return true 有
+     */
+    public Boolean hasNullField() {
+        return templateCode.isBlank() || validityPeriod == null ||
+                titleName.isBlank() || price == null || gradient == null ||
+                impactName == null || title.isBlank() || sColor.isBlank() || eColor.isBlank();
+    }
+
 }

--- a/src/main/java/cn/chahuyun/economy/manager/TitleManager.java
+++ b/src/main/java/cn/chahuyun/economy/manager/TitleManager.java
@@ -244,7 +244,7 @@ public class TitleManager {
                                 "你已经拥有 %s 称号!", template.getTitleName()));
                         return;
                     }
-                    if (EconomyUtil.minusMoneyToUser(sender, 10000)) {
+                    if (EconomyUtil.minusMoneyToUser(sender, template.getPrice())) {
                         if (addTitleInfo(userInfo, template.getTemplateCode())) {
                             subject.sendMessage(MessageUtil.formatMessageChain(message,
                                     "你以成功购买 %s 称号,有效期 %s ", template.getTitleName(),

--- a/src/main/java/cn/chahuyun/economy/plugin/TitleTemplateManager.java
+++ b/src/main/java/cn/chahuyun/economy/plugin/TitleTemplateManager.java
@@ -7,7 +7,6 @@ import cn.chahuyun.economy.entity.UserInfo;
 import cn.chahuyun.economy.entity.title.CustomTitle;
 import cn.chahuyun.economy.entity.title.TitleTemplate;
 import cn.chahuyun.economy.manager.TitleManager;
-import cn.hutool.core.bean.BeanUtil;
 import cn.hutool.core.date.DateUtil;
 import cn.hutool.core.io.FileUtil;
 import cn.hutool.json.JSONArray;
@@ -147,8 +146,8 @@ public final class TitleTemplateManager {
                 continue;
             }
 
-            if (BeanUtil.hasNullField(customTitle)) {
-                log.warn("自定义称号错误:{}", customTitle.getTitleName());
+            if (customTitle.hasNullField()) {
+                log.warn("自定义称号错误:{} ,所有属性必填!", customTitle.getTitleName());
                 continue;
             }
 

